beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -750,3 +758,327 @@ func (bs *Server) checkSync(ctx context.Context, w http.ResponseWriter) bool {
 	}
 	return true
 }
+
+// ProduceBlockV3 Requests a beacon node to produce a valid block, which can then be signed by a validator. The
+// returned block may be blinded or unblinded, depending on the current state of the network as
+// decided by the execution and beacon nodes.
+// The beacon node must return an unblinded block if it obtains the execution payload from its
+// paired execution node. It must only return a blinded block if it obtains the execution payload
+// header from an MEV relay.
+// Metadata in the response indicates the type of block produced, and the supported types of block
+// will be added to as forks progress.
+func (bs *Server) ProduceBlockV3(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ProduceBlockV3")
+	defer span.End()
+	if ok := bs.checkSync(r.Context(), w); !ok {
+		return
+	}
+	segments := strings.Split(r.URL.Path, "/")
+	rawSlot := segments[len(segments)-1]
+	rawRandaoReveal := r.URL.Query().Get("randao_reveal")
+	rawGraffiti := r.URL.Query().Get("graffiti")
+	rawSkipRandaoVerification := r.URL.Query().Get("skip_randao_verification")
+
+	slot, valid := shared.ValidateUint(w, "slot", rawSlot)
+	if !valid {
+		return
+	}
+
+	var randaoReveal []byte
+	if rawSkipRandaoVerification == "true" {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := hexutil.Decode(rawRandaoReveal)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode randao reveal").Error(), http.StatusBadRequest)
+			return
+		}
+		randaoReveal = rr
+	}
+	if len(randaoReveal) != fieldparams.BLSSignatureLength {
+		http2.HandleError(w, fmt.Sprintf("a valid randao reveal is required as a query parameter: received length %d but wanted length %d", len(randaoReveal), fieldparams.BLSSignatureLength), http.StatusBadRequest)
+		return
+	}
+	var graffiti []byte
+	if rawGraffiti != "" {
+		g, err := hexutil.Decode(rawGraffiti)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode graffiti").Error(), http.StatusBadRequest)
+			return
+		}
+		graffiti = g
+	}
+
+	produceBlockV3(ctx, bs, w, r, &eth.BlockRequest{
+		Slot:         primitives.Slot(slot),
+		RandaoReveal: randaoReveal,
+		Graffiti:     graffiti,
+		SkipMevBoost: false,
+	})
+}
+
+func produceBlockV3(context context.Context, bs *Server, w http.ResponseWriter, r *http.Request, v1alpha1req *eth.BlockRequest) {
+	ctx, span := trace.StartSpan(context, "validator.ProduceBlockV3.internal")
```

Comment:
Is this useful? Same comment for other spans in private functions

Reply:
hmm maybe not maybe I should just pass it down